source/MaterialXShaderGen/Syntax.h

```diff
@@ -12,95 +12,62 @@
 namespace MaterialX
 {
 
-class SgInput;
-class SgOutput;
-
 using SyntaxPtr = shared_ptr<class Syntax>;
+using TypeSyntaxPtr = shared_ptr<class TypeSyntax>;
 
-/// Base class for syntax objects used by code generators
-/// to emit code with correcy syntax for each language
+/// Base class for syntax objects used by shader generators
+/// to emit code with correcy syntax for each language.
 class Syntax
 {
 public:
-    /// Struct holding information on syntax to be used for data types
-    /// Required to be set for each data type
-    struct TypeSyntax
```

Comment:
Pulling this out and making it a proper base class, with derived implementations for various type syntaxes.